include/up-cpp/rpc/RpcClient.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * 
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef _RPC_CLIENT_H_
+#define _RPC_CLIENT_H_
+
+#include <future>
+#include <up-cpp/transport/UListener.h>
+#include <up-cpp/transport/datamodel/UMessage.h>
+#include <up-core-api/uattributes.pb.h>
+#include <up-core-api/ustatus.pb.h>
+
+namespace uprotocol::rpc {
+
+    struct RpcResponse {
+        uprotocol::v1::UStatus status;
+        uprotocol::utransport::UMessage message;
```

Comment:
I think we should find a way to NOT expose protobuf headers in the client API. Solving this problem would us much more freedom in the implementation. And, depending on how its done, much cleaner internal code.

Reply:
> I haven't yet heard a convincing argument on why the protobuf headers _should not_ be exposed as part of the client API. Is there a _specific_ problem we are trying to avoid? Is there a usability issue? This doesn't seem to be an issue in other language implementations.

I will say that when I was looking into how to package `up-cpp` for CCI that there were a lot of gripes / concerns from people about protobuf headers being exposed.

By doing so, you lock the user into needing to be on the same protobuf version, in-lockstep. Why is this bad? Well, the real concern was that by doing so, if you have "platform supported libraries" then you'd also have to bump the version of protobuf supported within that platform, incurring potentially a cascading change across other libraries which had used exposed protobuf headers.

In other languages, like Java and Rust, the end-user of `up-client-foo-java` or `up-client-foo-rust` is shielded from this _to an extent_, as they would only need to bump their version of the client library and get the new protobuf version "for free" due to the package manager handling this opaquely.

In C++, if the protobuf headers are exposed, then anyone _using_ `up-client-zenoh-cpp` needs to also bump the version, i.e. _every single uE written_.

Maybe we're willing to do that? But it does seem like a pain and could cause user error.

Reply:
Also note that I'm not buy or selling any particular alternative :slightly_smiling_face: 

Just pointing out a disadvantage of having protobuf headers exposed

Reply:
@PLeVasseur - That is an interesting point. The library dependencies could get strange if an application were using a different version of the protobuf library than our library is. While it would be reasonably possible to use private linking pull newer protobuf into the compiled up-cpp library without disturbing applications, the protobuf headers themselves aren't clean.

This will somewhat come down to how strongly @stevenhartley feels that the protobuf objects should be visible or not. We _could_ definitely hide them away such that users of up-_something_-cpp don't have to worry about it. Alternatively, we could make the protobuf library version a setting that can be passed to the conan recipe so that up-cpp can be built to match whatever the application is using. Or we can stay the course and require any application using up-cpp to match protobuf versions.

None of those options are nice.